File: Java/Best-Time-to-Buy-and-Sell-Stock-II.java
class Solution {
    /**
     * @param prices: Given an integer array
     * @return: Maximum profit
     */
    public int maxProfit(int[] prices) {
        // write your code here
        int res = 0, cur = Integer.MAX_VALUE;
        for (int i = 0; i < prices.length; i ++) {
            if (prices[i] >= cur) {
                res = res + (prices[i] - cur);
            }
            cur = prices[i];
        }
        return res;
    }
};

File: Cpp/Best-Time-to-Buy-and-Sell-Stock-II.cpp
#include <vector>

class Solution {
public:
    /**
     * @param prices: Given an integer array
     * @return: Maximum profit
     */
    int maxProfit(vector<int> &prices) {
        // write your code here
        if (prices.size() <= 1) return 0;
        int sum = 0, len = prices.size();
        int cur = prices[0];
        for (int i = 1; i < len; i++) {
            if (prices[i] < cur) {
                cur = prices[i];
            } else {
                sum += (prices[i] - cur);
                cur = prices[i];
            }
        }
        return sum;
    }
};